columnq/src/table/database.rs

```diff
@@ -73,23 +93,63 @@ mod imp {
                     }
                 }
                 DatabaseLoader::Postgres => {
-                    // ToDo `Cannot start a runtime from within a runtime` error in `connector-x PostgresSource`
-                    return Err(ColumnQError::Database(
-                        "Postgres database features not be supported for now.".to_string(),
-                    ));
+                    #[cfg(feature = "database-postgres")]
+                    {
+                        let config = tokio_postgres::Config::from_str(t.get_uri_str())
+                            .map_err(|e| ColumnQError::Database(e.to_string()))?;
+                        let tls = match config.get_ssl_mode() {
+                            tokio_postgres::config::SslMode::Require => tokio_postgres::NoTls,
+                            _ => tokio_postgres::NoTls,
+                        };
+                        let source: PostgresSource<BinaryProtocol, tokio_postgres::NoTls> =
+                            PostgresSource::new(config.into(), tls, 2)
+                                .map_err(|e| ColumnQError::Database(e.to_string()))?;
+                        let queries = queries.clone();
+                        let task = tokio::task::spawn_blocking(move || {
```

Comment:
I might be missing something here, why do we need to fetch postgres data with blocking code here?

Reply:
if we don't block there, will cause err: `Cannot start a runtime from within a runtime`.

Reply:
This is specific to the PostgresSource? I will take a closer look at it this weekend, thanks for the patch!

If this is expected, then we should also file an upstream bug report, async call context should be supported as first class citizen in connectorx.

Reply:
yes, it's specific to the `PostgresSource`.

Reply:
@holicc Do you know why have this error `Cannot start a runtime from within a runtime`?

Reply:
In somewhere create a new runtime when using PgSource